Git segment: Template rendering throws error for bare repos

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

When "in" a bare repo the template rendering is broken.

![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/24372132/b76b9d09-dd0e-4542-b6b7-16ba6f3c76ee)

Unfortunately, I don't know the version number this came up with. But I know it was working because I've requested this feature (#2913). ^^

Let me know how I can help further.

### Theme

```
{
    "background": "#1A1A1A",
    "foreground": "#da70d6",
    "leading_diamond": "<transparent,background>\uE0B0</>",
    "trailing_diamond": "\ue0b4",
    "properties": {
        "fetch_bare_info": true,
        "fetch_stash_count": true,
        "fetch_status": true,
        "fetch_upstream_icon": true,
        "fetch_worktree_count": true
    },
    "style": "diamond",
    "template": "{{ .UpstreamIcon }} {{ .HEAD }}{{if .IsBare }}, bare{{ end }}{{if .Working.String }} {{ .Working }}{{ end }}<#FFC0CB>{{if .Staging.Changed }}staged:{{ end }}{{if .Staging.Modified }} m{{ .Staging.Modified }}{{ end }}{{if .Staging.Added }} +{{ .Staging.Added }}{{ end }}{{if .Staging.Deleted }} -{{ .Staging.Deleted }}{{ end }}{{if .StashCount }}{{if .Staging.Changed }},{{ end }} stashed: {{ .StashCount }}{{ end }}</>",
    "type": "git"
}
```

### What OS are you seeing the problem on?

Windows

### Which shell are you using?

powershell

### Log output

```Shell
[ERROR] 19:55:45.862 text.go:Render:86 → template: {{ .Data.UpstreamIcon }} {{ .Data.HEAD }}{{if .Data.IsBare }}, bare{{ end }}{{if .Data.Working.String }} {{ .Data.Working }}{{ end }}<#FFC0CB>{{if .Data.Staging.Changed }}staged:{{ end }}{{if .Data.Staging.Modified }} m{{ .Data.Staging.Modified }}{{ end }}{{if .Data.Staging.Added }} +{{ .Data.Staging.Added }}{{ end }}{{if .Data.Staging.Deleted }} -{{ .Data.Staging.Deleted }}{{ end }}{{if .Data.StashCount }}{{if .Data.Staging.Changed }},{{ end }} stashed: {{ .Data.StashCount }}{{ end }}</>:1:86: executing "{{ .Data.UpstreamIcon }} {{ .Data.HEAD }}{{if .Data.IsBare }}, bare{{ end }}{{if .Data.Working.String }} {{ .Data.Working }}{{ end }}<#FFC0CB>{{if .Data.Staging.Changed }}staged:{{ end }}{{if .Data.Staging.Modified }} m{{ .Data.Staging.Modified }}{{ end }}{{if .Data.Staging.Added }} +{{ .Data.Staging.Added }}{{ end }}{{if .Data.Staging.Deleted }} -{{ .Data.Staging.Deleted }}{{ end }}{{if .Data.StashCount }}{{if .Data.Staging.Changed }},{{ end }} stashed: {{ .Data.StashCount }}{{ end }}</>" at <.Data.Working.String>: error calling String: runtime error: invalid memory address or nil pointer dereference
```

--- a/src/segments/git.go
+++ b/src/segments/git.go
@@ -167,6 +167,9 @@ func (g *Git) Enabled() bool {
 
 	g.RepoName = platform.Base(g.env, g.convertToLinuxPath(g.realDir))
 
+	g.Working = &GitStatus{}
+	g.Staging = &GitStatus{}
+
 	if g.IsBare {
 		g.getBareRepoInfo()
 		return true
@@ -186,8 +189,6 @@ func (g *Git) Enabled() bool {
 		g.setBranchStatus()
 	} else {
 		g.setPrettyHEADName()
-		g.Working = &GitStatus{}
-		g.Staging = &GitStatus{}
 	}
 	if g.props.GetBool(FetchUpstreamIcon, false) {
 		g.UpstreamIcon = g.getUpstreamIcon()

